bitcoind_mock/transaction.py

```diff
@@ -1,152 +1,20 @@
-                # There is some error in the serialized transaction passed as input. Transaction can't be built
-                tx = None
-            else:
-                tx.offset = 0
-
-        except ValueError:
-            # If a parsing error occurs, the deserialization stops and None is returned
-            tx = None
-
-        return tx
-
-    def serialize(self, rtype=hex):
-        """ Serialize all the transaction fields arranged in the proper order, resulting in a hexadecimal string
-        ready to be broadcast to the network.
-        :param self: self
-        :type self: TX
-        :param rtype: Whether the serialized transaction is returned as a hex str or a byte array.
-        :type rtype: hex or bool
-        :return: Serialized transaction representation (hexadecimal or bin depending on rtype parameter).
-        :rtype: hex str / bin
-        """
-
-        if rtype not in [hex, bin]:
-            raise Exception("Invalid return type (rtype). It should be either hex or bin.")
-        serialized_tx = change_endianness(int2bytes(self.version, 4))  # 4-byte version number (LE).
-
-        # INPUTS
-        serialized_tx += encode_varint(self.inputs)  # Varint number of inputs.
-
-        for i in range(self.inputs):
-            serialized_tx += change_endianness(self.prev_tx_id[i])  # 32-byte hash of the previous transaction (LE).
-            serialized_tx += change_endianness(int2bytes(self.prev_out_index[i], 4))  # 4-byte output index (LE)
-            serialized_tx += encode_varint(len(self.scriptSig[i]) // 2)  # Varint input script length.
-            # ScriptSig
-            serialized_tx += self.scriptSig[i]  # Input script.
-            serialized_tx += int2bytes(self.nSequence[i], 4)  # 4-byte sequence number.
-
-        # OUTPUTS
-        serialized_tx += encode_varint(self.outputs)  # Varint number of outputs.
-
-        if self.outputs != 0:
-            for i in range(self.outputs):
-                serialized_tx += change_endianness(int2bytes(self.value[i], 8))  # 8-byte field Satoshi value (LE)
-                # ScriptPubKey
-                serialized_tx += encode_varint(len(self.scriptPubKey[i]) // 2)  # Varint Output script length.
-                serialized_tx += self.scriptPubKey[i]  # Output script.
-
-        serialized_tx += int2bytes(self.nLockTime, 4)  # 4-byte lock time field
-
-        # If return type has been set to binary, the serialized transaction is converted.
-        if rtype is bin:
-            serialized_tx = unhexlify(serialized_tx)
-
-        return serialized_tx
+from bitcoind_mock.utils import *
 
-    @staticmethod
-    def create_dummy_transaction(prev_tx_id=None, prev_out_index=None):
-        tx = TX()
 
-        if prev_tx_id is None:
-            prev_tx_id = urandom(32).hex()
+def create_dummy_transaction(prev_tx_id=None, prev_out_index=None):
+    if prev_tx_id is None or len(prev_tx_id) != 64:
+        prev_tx_id = urandom(32).hex()
```

Comment:
I think I did this because you use `urandom(32)` in other parts of the codebase. will break it out into a utility function. should I replace `urandom` in `get_new_block_data` in `bitcoind.py`?

Reply:
Nevermind, forget the comment, it is not relevant here actually. 

I tend to use this approach for testing and got it messed up here 😓

Reply:
deterministic behavior seems like a reasonable goal for a mock. may be nice to just use a seeded RNG everywhere in the long run :thinking:

Reply:
Fair enough.

Reply:
> I think I did this because you use `urandom(32)` in other parts of the codebase. will break it out into a utility function. should I replace `urandom` in `get_new_block_data` in `bitcoind.py`?

Sorry I didn't reply to this. Yes it would be nice for consistency to have it all match.

Reply:
Should I do this now or just leave it?

Reply:
Change it if you don't mind.

Reply:
done